file-formats/fugr/type/zif_fb_aff_func_v1.intf.abap

```diff
@@ -0,0 +1,246 @@
+      start_immediately_no_restart TYPE ty_update_task_kind VALUE '3',
+      "! <p class="shorttext">Collective Run</p>
+      "! Collective Run
+      collective_run               TYPE ty_update_task_kind VALUE '5',
+      "! <p class="shorttext">Unsupported Kind</p>
+      "! Unsupported Kind
+      unsupported_kind             TYPE ty_update_task_kind VALUE '6',
+    END OF co_update_task_kind.
+
+  "! <p class="shorttext">Release State</p>
+  "! Release state
+  "! $values {@link zif_fb_aff_func_v1.data:co_release_state}
+  TYPES ty_release_state TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Release State</p>
+    "! Release state
+    BEGIN OF co_release_state,
+      "! <p class="shorttext">Function module is released</p>
+      "! Function module is released
+      released              TYPE ty_release_state VALUE 'E',
+      "! <p class="shorttext">Function module is released SAP-internally</p>
+      "! Function module is released SAP-internally
+      released_sap_internal TYPE ty_release_state VALUE 'I',
+      "! <p class="shorttext">Function module is obsolete/deprecated</p>
+      "! Function module is obsolete/deprecated
+      obsolete              TYPE ty_release_state VALUE 'O',
+      "! <p class="shorttext">Function module is planned to be released [obsolete]</p>
+      "! Function module is planned to be released [obsolete]
+      release_planned       TYPE ty_release_state VALUE 'M',
+      "! <p class="shorttext">Function module is not released</p>
+      "! Function module is not released
+      not_released          TYPE ty_release_state VALUE 'N',
+    END OF co_release_state.
+
+  TYPES:
+    "! <p class="shorttext">Update specific fields</p>
+    "! Specific fields for update function modules
+    BEGIN OF ty_update_properties,
+      "! <p class="shorttext">Update task kind</p>
+      "! Update task kind
+      "! $required
+      update_task_kind TYPE ty_update_task_kind,
+    END OF ty_update_properties.
+
+
+  TYPES:
+    "! <p class="shorttext">Object Type FUNC</p>
+    "! FUNC object type
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Schema</p>
+      "! Format version
+      "! $required
+      schema               TYPE string,
+      "! <p class="shorttext">Description</p>
+      "! Description of a function module
+      "! $required
+      description          TYPE zif_aff_types_v1=>ty_description_60,
+      "! <p class="shorttext">Processing Type</p>
+      "! Processing type
+      "! $required
+      processing_type      TYPE ty_processing_type,
+      "! <p class="shorttext">RFC specific fields</p>
+      "! Specific fields for rfc function modules
+      rfc_properties       TYPE ty_rfc_properties,
+      "! <p class="shorttext">Update specific fields</p>
+      "! Specific fields for update function modules
+      update_properties    TYPE ty_update_properties,
+      "! <p class="shorttext">Release state</p>
+      "! Release state
+      release_state        TYPE c LENGTH 1,
```

Comment:
Can we specify the allowed values in a constant? Isn't it the following type?

```suggestion
      release_state        TYPE ty_release_state,
```

Reply:
That's right. I didn't see that there ist already a type for this